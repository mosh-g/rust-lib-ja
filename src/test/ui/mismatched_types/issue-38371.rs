// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

struct Foo {
}

fn foo(&foo: Foo) {  // illegal syntax
}

fn bar(foo: Foo) {  // legal
}

fn qux(foo: &Foo) {  // legal
}

fn zar(&foo: &Foo) {  // legal
}

fn main() {}
