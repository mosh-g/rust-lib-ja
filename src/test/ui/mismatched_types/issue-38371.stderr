error[E0308]: mismatched types
  --> $DIR/issue-38371.rs:16:8
   |
16 | fn foo(&foo: Foo) {
   |        ^^^^ expected struct `Foo`, found reference
   |
   = note: expected type `Foo`
   = note:    found type `&_`
   = help: did you mean `foo: &Foo`?

error[E0308]: mismatched types
  --> $DIR/issue-38371.rs:28:9
   |
28 | fn agh(&&bar: &u32) {
   |         ^^^^ expected u32, found reference
   |
   = note: expected type `u32`
   = note:    found type `&_`

error[E0529]: expected an array or slice, found `u32`
  --> $DIR/issue-38371.rs:31:9
   |
31 | fn ugh(&[bar]: &u32) {
   |         ^^^^^ pattern cannot match with input type `u32`

error: aborting due to 3 previous errors

